07_leukemia_patient/workflow/workflow.sh

```diff
@@ -0,0 +1,232 @@
+
+cd $WD
+
+# for mapped reads: they will have the gx tag, unmapped won't so will have different column number
+
+# getting reads in BCR and ABL region and adding 1000 bp on each side of the region
+
+bcr_start=23179704
+bcr_end=23318037
+abl_start=130713016
+abl_end=130887675
+
+echo "getting read names"
+
+touch r2.fastq
+
+samtools view -H $STARSOLO_BAM > header.txt
+
+samtools view $STARSOLO_BAM "chr22:22179704-24318037" | cat header.txt - | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$27";"$28"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+samtools view $STARSOLO_BAM "chr9:129713016-131887675" | cat header.txt - | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$27";"$28"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+# for unmapped reads: they don't have the gx tag, so the CB and UB will be different
+
+samtools view -f 4 $STARSOLO_BAM | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$22";"$23"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+echo "finished getting read names"
+
+rm header.txt
+
+# deduplicate .fastqs
+
+cat r2.fastq | paste - - - - | sort -k1,1 -t " " | uniq | tr "\t" "\n" > sorted_duplicate_r2.fastq
+
+pigz sorted_duplicate_r2.fastq
+
+rm r2.fastq
+
+echo "R2 file generated"
+
+# run bwa_mem2
+
+mkdir -p $WD/bwa_mem2/genome
+cd $WD/bwa_mem2/genome
+
+# genome generation
+
+wget http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/"$TRANSCRIPTOME".gz
+
+pigz --decompress *gz
+
+grep "ABL1" "$TRANSCRIPTOME"| sed 's/>//g' > ABL1_human.gtf
+grep "BCR" "$TRANSCRIPTOME"| sed 's/>//g' >  BCR_human.gtf
+
+touch ABL1_human.fa
+touch BCR_human.fa
+
+for i in $(cat ABL1_human.gtf); do
+    samtools faidx $TRANSCRIPTOME "$i" >> ABL1_human.fa
+done
+
+for i in $(cat BCR_human.gtf) ; do
+    samtools faidx $TRANSCRIPTOME "$i"  >> BCR_human.fa
+done
+
+cat "$CUSTOM_FA" ABL1_human.fa BCR_human.fa > combined.fa
+cat "$CUSTOM_GTF" ABL1_human.gtf BCR_human.gtf > combined.gtf
+
+# index
+
+~/leukemia_bwamem2/bwa-mem2-2.2.1_x64-linux/bwa-mem2 index -p indexed combined.fa
+
+echo "BWAMEM2 finished"
+
+# run
+
+mkdir -p $WD/bwa_mem2/output
+cd $WD/bwa_mem2/output
+
+~/leukemia_bwamem2/bwa-mem2-2.2.1_x64-linux/bwa-mem2 mem $WD/bwa_mem2/genome/indexed $WD/sorted_duplicate_r2.fastq.gz -t 20 -k 80 -w 12 | samtools view -bS | samtools sort -o bwa_mem2.sorted.bam 
+
+# run STAR fusion --> the genome takes too long to generate (3 days), use premade genome
```

Comment:
i don't think this is safe, unless the premade genome contains the exact human genome version and annotation (gtf and txome - perhaps gencode)

Reply:
I think it does yes!

Reply:
can you treport the premade genome annotation version? and genome assembly with patch?